lua/impatient.lua

```diff
@@ -33,7 +36,7 @@ local function load_mpack()
   return require('mpack')
 end
 
-local mpack = load_mpack()
+local mpack = (_G.use_cachepack and require('impatient.cachepack')) or load_mpack()
```

Comment:
On my machine (M1 Macbook), mpack appears to be a slightly quicker on average. Maybe we should use cachepack as a fallback for users who don't have mpack?

Reply:
I usually see it within a difference of 1ms, which I'm betting is more due to context switching than anything.
On Linux I tested both about 20 times and most of the time reads were within that tolerance, but occasionally cachepack was even faster :shrug:
Almost every time however, writes were ~10ms faster with cachepack.

How much quicker was mpack?

Here's from the tests on the this repo by the way:
`v0.50`
```
[ RUN      ] impatient load plugins without impatient: 2936.55 ms OK
[ RUN      ] impatient run without cache creates cache using mpack: 77.92 ms OK
[ RUN      ] impatient run without cache creates cache using cachepack: 65.41 ms OK
[ RUN      ] impatient run with cache using mpack loads: 49.81 ms OK
[ RUN      ] impatient run with cache using cachepack loads: 50.68 ms OK
```
`nightly`
```
[ RUN      ] impatient load plugins without impatient: 2690.13 ms OK
[ RUN      ] impatient run without cache creates cache using mpack: 66.97 ms OK
[ RUN      ] impatient run without cache creates cache using cachepack: 56.84 ms OK
[ RUN      ] impatient run with cache using mpack loads: 43.23 ms OK
[ RUN      ] impatient run with cache using cachepack loads: 41.87 ms OK
```

Reply:
I measured the times using `:LuaCacheProfile` which shouldn't be affected by much.

The difference I noticed was pretty negligible (<1ms).

I ran again a few more times to double-check, and to be honest, cachepack could well be quicker, so let's just keep it as it is.

In a future change, I'll hook up the tests to report the time reported by `:LuaCacheProfile` since it should be more precise.